frontend/src/features/myinfo/utils/augmentWithMyInfo.ts

```diff
@@ -3,22 +3,31 @@ import { keyBy } from 'lodash'
 import { types as myInfoTypeArray } from '~shared/constants/field/myinfo'
 import { BasicField, FormFieldDto, MyInfoFormField } from '~shared/types/field'
 
+import { isMyInfo } from './isMyInfo'
+
 const MAP_ATTR_TO_NAME = keyBy(myInfoTypeArray, 'name')
 
 // Making a copy by destructuring so original object does not get affected.
 export const augmentWithMyInfo = ({
   ...field
 }: FormFieldDto): MyInfoFormField => {
-  // Only dropdown fields have augmented options for now.
+  if (!isMyInfo(field)) return field
+
+  const myInfoBlock = MAP_ATTR_TO_NAME[field.myInfo.attr]
+
   switch (field.fieldType) {
     case BasicField.Dropdown: {
-      // No need to augment if no MyInfo attribute
-      if (!field.myInfo?.attr) return field
-      const myInfoBlock = MAP_ATTR_TO_NAME[field.myInfo.attr]
-      field.fieldOptions = myInfoBlock.fieldOptions ?? []
-      return field
+      return {
+        ...field,
+        fieldValue: myInfoBlock.prefilledValue,
```

Comment:
suggest not changing this function, since it may be confusing as to when `prefilledValue` may exist on public forms.

Reply:
yep, i think changing to injecting the default values thru the initial `useForm` call would eliminate the need for this